### PR TITLE
fix/torchmodel_prediction_large_n

### LINKED ISSUE #673: [BUG] PastCovariatesTorchModel prediction fails with n > output_chunk_length
**Describe the bug**
PastCovariatesTorchModels (such as `NBEATSModel`) prediction fails when `n > output_chunk_length` AND `n` is not a multiple of `output_chunk_length`.

This comes down to `TorchForecastingModel._get_batch_prediction()`

**To Reproduce**
```
from darts.models import NBEATSModel
from darts.datasets import AirPassengersDataset
from darts.utils.timeseries_generation import linear_timeseries

series = AirPassengersDataset().load()
covs = linear_timeseries(start=series.start_time(), length=len(series) + 30, freq=series.freq)
model = NBEATSModel(input_chunk_length=10, output_chunk_length=10, n_epochs=1)
model.fit(series, past_covariates=covs)
pred = model.predict(n=13, past_covariates=covs)
```

**Expected behavior**
Predictions should work for any `n`.

**System (please complete the following information):**
 - Python version: 3.7.11
 - darts version 0.14.0

### PR DESCRIPTION
**Pull request by dennisbader**
PastCovariatesTorchModels (such as NBEATSModel) prediction failed when `n > output_chunk_length` AND `n` not being a multiple of `output_chunk_length`.

Fixes #673.

### Summary

- refactored `_get_batch_prediction()` for PastCovariatesTorchModel by taking TFTModels implementation for past covariates.
- added unit tests
- fixed deprecation warning in TFTModel
- added TFTModel to global forecasting model unit tests

### CODE REVIEW

File: darts/models/forecasting/tft_model.py [hunk 1]
```diff
@@ -288,7 +288,7 @@ def get_relative_index(encoder_length: int,
         prediction_index = encoder_length - 1
         index[:encoder_length] = index[:encoder_length] / prediction_index
         index[encoder_length:] = index[encoder_length:] / prediction_index
-        return index.resize(1, len(index), 1).repeat(batch_size, 1, 1)
```
Comment (hrzn) [inline darts/models/forecasting/tft_model.py:291]: Wait, this was returning None?

File: darts/tests/test_global_forecasting_models.py [hunk 2]
```diff
@@ -218,6 +222,30 @@ def test_predict_from_dataset_unsupported_input(self):
...
+                    past_covs, future_covs = self.covariates, None
+                elif isinstance(model, DualCovariatesTorchModel):
+                    past_covs, future_covs = None, self.covariates
+                else:
+                    past_covs, future_covs = self.covariates, self.covariates
+
+                model.fit(self.target_past, past_covariates=past_covs, future_covariates=future_covs, epochs=1)
+
+                # test prediction for n < out_len, n == out_len and n > out_len
+                for n in [OUT_LEN - 1, OUT_LEN, 2 * OUT_LEN - 1]:
```
Comment (hrzn) [inline darts/tests/test_global_forecasting_models.py:245]: 💯

**Review by hrzn — APPROVED** [review]
Thanks!

File: darts/models/forecasting/tft_model.py (hunk 1 above)
Comment (dennisbader) [reply darts/models/forecasting/tft_model.py:291]: No, it just had a DeprecationWarning that non-inplace `resize` shouldn't be used anymore

**Pull request merged**